### PR TITLE
#43 ログイン機能追加

### LINKED ISSUE #43: サインインAPI作成

### PR DESCRIPTION
**Pull request by Sorarinu**
Close #43

### CODE REVIEW

File: server/api/app/Http/Controllers/Api/ApiController.php [hunk 1]
```diff
@@ -59,6 +60,35 @@ public function signUp()
...
+            if ($validation->passes()) {
+                $data = json_decode(json_encode([
+                    'password' => $this->request->input('password'),
+                    'email' => $this->request->input('email')
+                ]));
+
+                $user = User::where('email', $data->email)
+                    ->firstOrFail();
+                
+                if(Hash::check($data->password, $user->password)) {
```
Comment (Sorarinu) [inline server/api/app/Http/Controllers/Api/ApiController.php]: **[code_sniffer]**
- Expected 1 space after IF keyword; 0 found

File: server/api/app/Http/Controllers/Api/ApiController.php [hunk 2]
```diff
@@ -59,6 +60,35 @@ public function signUp()
...
+                    return response()->json(['status' => 'OK', 'message' => 'Login success: ' . $user->email]);
+                }
+
+                throw new \Exception;
+            }
+            
+            return response()->json(['status' => 'NG', 'message' => $validation->messages()]);
+
```
Comment (Sorarinu) [inline server/api/app/Http/Controllers/Api/ApiController.php:92]: **[code_sniffer]**
- Blank line found at end of control structure

File: server/api/app/Http/Controllers/Api/ApiController.php [hunk 3]
```diff
@@ -59,6 +60,35 @@ public function signUp()
...
+        }catch(\Exception $e) {
```
Comment (Sorarinu) [inline server/api/app/Http/Controllers/Api/ApiController.php]: **[code_sniffer]**
- Expected 1 space after closing brace; 0 found
- Expected 1 space after CATCH keyword; 0 found

File: server/api/app/Http/routes.php [hunk 4]
```diff
@@ -16,7 +16,7 @@
     });
 
     Route::post('api/v1/signup',                 ['uses' => 'api\ApiController@signUp']);
-    Route::get('api/v1/signin',                  ['uses' => 'api\ApiController@signIn']);
+    Route::post('api/v1/signin',                 ['uses' => 'api\ApiController@signIn']);
```
Comment (Sorarinu) [inline server/api/app/Http/routes.php]: **[code_sniffer]**
- Expected 1 space after comma in function call; 17 found

**Comment by Sorarinu** [comment]
Click here if you want to create code_sniffer's auto-correct Pull Request 
[Auto Correct by SideCI](https://sideci.com/gh/63405502/issues/pull_request/49/diff/3c114cff08e2731e11a30c9f95284d875c7b8c11?open_modal=1#code_sniffer)

See the documentation if you want to get more information about auto-correct.
[Document](https://sideci.com/ja/docs/autofix-code_sniffer)

**Comment by Sorarinu** [comment]
Created Auto-correct Pull Request from [SideCI](https://sideci.com/)

[https://github.com/Sorarinu/ProjectP2016_F/pull/50](https://github.com/Sorarinu/ProjectP2016_F/pull/50)

File: server/api/app/Http/Controllers/Api/ApiController.php [hunk 5]
```diff
@@ -59,6 +60,35 @@ public function signUp()
...
+                    ->firstOrFail();
+                
+                if(Hash::check($data->password, $user->password)) {
+                    return response()->json(['status' => 'OK', 'message' => 'Login success: ' . $user->email]);
+                }
+
+                throw new \Exception;
+            }
+            
+            return response()->json(['status' => 'NG', 'message' => $validation->messages()]);
```
Comment (namazu510) [inline server/api/app/Http/Controllers/Api/ApiController.php]: valodationの失敗時応答メッセージは単純な文字列でおねがいします
これってＪＳＯＮObjectだよね
あとサインアップのほうも多分オブジェクトが。。。

Comment (namazu510) [reply server/api/app/Http/Controllers/Api/ApiController.php]: プロパティmessageの中身が文字列ってことです

**Comment by namazu510** [comment]
Linterの設定を変えるか治すかして怒られないようしてください(T_T)

File: server/api/app/Http/routes.php [hunk 6]
```diff
@@ -1,25 +1,25 @@
...
-    Route::put('api/v1/bookmarks/{bookmark_id}', ['uses' => 'api\ApiController@update']);
-    Route::get('api/v1/bookmarks',               ['uses' => 'api\ApiController@getAll']);
\ No newline at end of file
+Route::post('api/v1/signup', ['uses' => 'api\ApiController@signUp']);
+Route::post('api/v1/signin', ['uses' => 'api\ApiController@signIn']);
+Route::get('api/v1/signout', ['uses' => 'api\ApiController@signOut']);
+Route::post('api/v1/bookmarks/upload', ['uses' => 'api\ApiController@upload']);
+Route::get('api/v1/bookmarks/export', ['uses' => 'api\ApiController@export']);
+Route::post('api/v1/bookmarks', ['uses' => 'api\ApiController@create']);
+Route::put('api/v1/bookmarks/{bookmark_id}', ['uses' => 'api\ApiController@update']);
```
Comment (Sorarinu) [inline server/api/app/Http/routes.php:24]: **[code_sniffer]**
- Expected 1 newline at end of file; 0 found

**Comment by Sorarinu** [comment]
Click here if you want to create code_sniffer's auto-correct Pull Request 
[Auto Correct by SideCI](https://sideci.com/gh/63405502/issues/pull_request/49/diff/9772478183861e3bdbf941ad74b0698990c69926?open_modal=1#code_sniffer)

See the documentation if you want to get more information about auto-correct.
[Document](https://sideci.com/ja/docs/autofix-code_sniffer)

**Comment by Sorarinu** [comment]
SideCIに勝利した

**Pull request merged**